### PR TITLE
Migrate to arrow-go

### PR DESCRIPTION
**Pull request by haoxins**
### What changes were proposed in this pull request?

The Arrow Go has its own repo now, see
https://github.com/apache/arrow/pull/44293

### Does this PR introduce _any_ user-facing change?

No

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -15,43 +15,43 @@
 
 module github.com/apache/spark-connect-go/v35
 
-go 1.21
+go 1.22.0
```
Comment (grundprinzip) [inline go.mod:18]: Is this change needed? Do we need to update the go version at the same time?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -115,5 +115,5 @@ jobs:
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v6
         with:
-          version: v1.59
```
Comment (grundprinzip) [inline .github/workflows/build.yml:118]: Same here.

**Review by grundprinzip — COMMENTED** [review]
Can you please decuple the updates of the go version from the rest of the changes? Thanks!

File: go.mod (hunk 1 above)
Comment (haoxins) [reply go.mod:18]: https://github.com/apache/arrow-go/blob/main/go.mod#L21

we need ~

**Comment by haoxins** [comment]
Wow, [v18.0.0](https://github.com/apache/arrow-go/releases/tag/v18.0.0) is released formally.

**Comment by grundprinzip** [comment]
Looks good, I tested locally as well. Thanks a lot!

**Review by grundprinzip — APPROVED** [review]

**Comment by grundprinzip** [comment]
merging to master
